lib/rules/no-get-with-default.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const types = require('../utils/types');
+
+function makeErrorMessageForGetWithDefault(property) {
+  return `Use \`get\` and \`||\` instead of \`getWithDefault\` for property \`${property}\``;
+}
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Use the || operator instead of getWithDefault for more expected behaviors',
+      category: 'Best Practices',
+      recommended: false,
+      octane: true,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-get-with-default.md',
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: false,
+      },
+    ],
+  },
+  create(context) {
+    return {
+      CallExpression(node) {
+        if (
+          types.isMemberExpression(node.callee) &&
+          types.isThisExpression(node.callee.object) &&
+          types.isIdentifier(node.callee.property) &&
+          node.callee.property.name === 'getWithDefault'
```

Comment:
Need to also check `node.arguments.length === 1`.

Reply:
Good point, but wouldn't this be 2? `this.getWithDefault('a', 'b')`

Reply:
Yes, you're right.